js/adapt-assessmentArticleModel.js

```diff
@@ -253,40 +251,31 @@ define([
       }
 
       if (!_.isEmpty(newSettings)) {
-        for (var i = 0, l = this._currentQuestionComponents.length; i < l; i++) {
-          this._currentQuestionComponents[i].set(newSettings, { pluginName: '_assessment' });
-        }
+        const questionComponents = this._getAllQuestionComponents();
+        questionComponents.forEach(model => model.set(newSettings, { pluginName: '_assessment' }));
       }
     },
 
     _setupQuestionListeners: function() {
-      var questionComponents = this._currentQuestionComponents;
-      for (var i = 0, l = questionComponents.length; i < l; i++) {
-        var question = questionComponents[i];
-        if (question.get('_isInteractionComplete')) continue;
-        this.listenTo(question, 'change:_isInteractionComplete', this._onQuestionCompleted);
-      }
+      this._removeQuestionListeners();
+      this.listenTo(this.getChildren(), 'change:_isInteractionComplete', this._onBlockCompleted);
     },
 
     _checkNumberOfQuestionsAnswered: function() {
-      var questionComponents = this._currentQuestionComponents;
-      var numberOfQuestionsAnswered = 0;
-      for (var i = 0, l = questionComponents.length; i < l; i++) {
-        var question = questionComponents[i];
-        if (question.get('_isInteractionComplete')) {
-          numberOfQuestionsAnswered++;
-        }
-      }
+      const questionComponents = this._getCurrentQuestionComponents();
+      const numberOfQuestionsAnswered = questionComponents.filter(model => model.get('_isInteractionComplete')).length;
       this.set('_numberOfQuestionsAnswered', numberOfQuestionsAnswered);
     },
 
     _removeQuestionListeners: function() {
-      var questionComponents = this._currentQuestionComponents;
-      if (!questionComponents) return;
-      for (var i = 0, l = questionComponents.length; i < l; i++) {
-        var question = questionComponents[i];
-        this.stopListening(question, 'change:_isInteractionComplete', this._onQuestionCompleted);
-      }
+      this.stopListening(this.getChildren(), 'change:_isInteractionComplete', this._onBlockCompleted);
+    },
+
+    _onBlockCompleted: function(blockModel, value) {
+      if (value === false) return;
+      var questionModel = blockModel.findDescendantModels('question')[0];
```

Comment:
Not often configured this way, but if there is more than one question in a block, the assessment won't complete due to this line.

Reply:
fixed